web/packages/teleport/src/Console/DocumentKubeExec/DocumentKubeExec.tsx

```diff
@@ -67,7 +71,24 @@ export default function DocumentKubeExec({ doc, visible }: Props) {
           errorText={webauthn.errorText}
         />
       )}
-      {status === 'initialized' && terminal}
+
+      {status === 'initialized' && !sentData && (
```

Comment:
I feel like it might be a good idea to introduce an explicit status instead of using an auxiliary boolean variable.

`DocumentSsh` has two statuses, `initialized` and `loading`. `DocumentKubeExec` at the moment has three, `initialized`, `loading`, `disconnected`.

What do you think about introducing a new status called `waiting-for-exec-data`? This way where `useKubeExecSession` now calls `setStatus('initialized')`, it'd call `setStatus('waiting-for-exec-data`)` instead.

Instead of passing an arbitrary `onExec` callback to `KubeExecData`, `useKubeExecSession` could return a function called `sendKubeExecData` that calls the corresponding method on `tty` and switches the status to `initialized`.

Long term I think it'd be easier for someone new to understand how different statuses map to actual behavior of the component, as it'd be a matter of understanding four statuses vs three + how `sentData` interacts with them.

Reply:
Yep, it's cleaner that way and we get rid of "sentData", I've changed it 👍 https://github.com/gravitational/teleport/pull/41466/commits/32131809293420e45a31c3ee88cd50d22e6173ec